catalog/plugin/catalog-plugin-metadatabackup/src/main/java/org/codice/ddf/catalog/plugin/metadata/backup/MetadataBackupPlugin.java

```diff
@@ -0,0 +1,313 @@
+        }
+
+        Object keepDeletedMetacards = properties.get(KEEP_DELETED_METACARDS_PROPERTY);
+        if (keepDeletedMetacards instanceof Boolean) {
+            this.keepDeletedMetacards = (Boolean) keepDeletedMetacards;
+            LOGGER.debug("Updating {} with {}", KEEP_DELETED_METACARDS_PROPERTY,
+                    keepDeletedMetacards);
+        }
+
+        Object folderDepth = properties.get(FOLDER_DEPTH_PROPERTY);
+        if (folderDepth instanceof Integer) {
+            setFolderDepth((Integer) folderDepth);
+            LOGGER.debug("Updating {} with {}", FOLDER_DEPTH_PROPERTY, folderDepth);
+        }
+
+    }
+
+    private void processRequest(ProcessRequest<? extends ProcessResourceItem> processRequest)
+            throws PluginExecutionException {
+        LOGGER.trace("Backing up metadata");
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new PluginExecutionException(
+                    "Unable to backup ingested metadata; no outputDirectory.");
+        }
+
+        if (metacardTransformer == null) {
+            throw new PluginExecutionException(
+                    "Unable to backup ingested metadata; no Metacard Transformer found.");
+        }
+
+        List<? extends ProcessResourceItem> processResourceItems = processRequest.getProcessItems();
+        for (ProcessResourceItem processResourceItem : processResourceItems) {
+            Metacard metacard = processResourceItem.getMetacard();
+            try {
+                LOGGER.trace("Backing up metadata for metacard : {}", metacard.getId());
+                BinaryContent binaryContent = metacardTransformer.transform(metacard, null);
+                copyMetadataToOutputDirectory(binaryContent, metacard.getId());
+            } catch (CatalogTransformerException e) {
+                LOGGER.debug("Unable to transform metacard with id {}.", metacard.getId(), e);
+            }
+        }
+    }
+
+    private void copyMetadataToOutputDirectory(BinaryContent content, String metacardId) {
+        if (content == null || content.getInputStream() == null) {
+            LOGGER.debug("No content for transformed metacard {}.", metacardId);
+            return;
+        }
+
+        File parent = getCompleteDirectory(getFolderDepth(), metacardId);
+        if (!parent.exists()) {
+            try {
+                FileUtils.forceMkdir(parent);
+            } catch (IOException e) {
+                LOGGER.debug("Unable to make directory {}", parent.getPath(), e);
+            }
+        }
+
+        validateDirectory(parent);
+        File file = new File(parent.getPath() + File.separator + metacardId);
+        LOGGER.trace("Writing metadata from {} to file {}", metacardId, file.getPath());
+
+        try (OutputStream outputStream = new FileOutputStream(file)) {
+            IOUtils.write(content.getByteArray(), outputStream);
+        } catch (IOException e) {
+            LOGGER.debug("Unable to backup {}. ", metacardId, e);
```

Comment:
Should this log level be higher? This could be thrown in the scenario the directory runs out of room/etc which is something an admin could take action on.